plugins/dialog/plugin.js

```diff
@@ -500,6 +503,150 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 
 		this.changeFocus = changeFocus;
 
+		function getCurrentIndex( startIndex, currentIndex, offset ) {
+			if ( startIndex < 0 ) {
+				startIndex = 0;
+			}
+
+			var dialogUiElements = me._.focusList,
+				direction = offset === 1 ? 'next' : 'prev',
+				isPreviousNodeInRadioGroup =
+					dialogUiElements[ startIndex ].role &&
+					dialogUiElements[ startIndex ].role === 'radiogroup',
+				isCurrentNodeInRadioGroup =
+					dialogUiElements[ currentIndex ].role &&
+					dialogUiElements[ currentIndex ].role === 'radiogroup';
+
+			// If the previous and the current element is not in the radio group or if the current
+			// element pointing outside the radio group the return the current index.
+			if (
+				( !isCurrentNodeInRadioGroup && !isPreviousNodeInRadioGroup ) ||
+				( !isCurrentNodeInRadioGroup && isPreviousNodeInRadioGroup )
+			) {
+				return currentIndex;
+			}
+
+			// If the current index points to the radio group, find the checked element and
+			// move focus to it.
+			if ( !isPreviousNodeInRadioGroup && isCurrentNodeInRadioGroup ) {
+				var radioGroup = getRadioGroup( currentIndex, direction ),
+					checkedItemIndex = getFocusedRadioGroupElement( radioGroup );
+
+				// If no radio button is checked then:
+				// When Tab was pressed, then select the first radio button.
+				// When Shift+Tab was pressed, then select the last radio button.
+				if ( checkedItemIndex === -1 ) {
+					checkedItemIndex = 0;
+					dialogUiElements[ currentIndex ].getInputElement().$.checked = true;
+				}
```

Comment:
This logic should not check the first/last item in the radio group but just focus it. Focus is a separate concept from checking. Focus basically means that the given element can be now operated by keyboard, checking changes the state of the element.